src/components/Dashboard/TodoList.vue

```diff
@@ -796,6 +833,70 @@ export default class TodoList extends Mixins(AllowableActionsMixin, DateMixin, E
     }
   }
 
+  /** check if the item is actually affiliation invite todo; (not a regular filing item). */
+  isAffiliationInvitation (item): boolean {
+    return item.status === FilingStatus.AFFILIATION_INVITATION_PENDING
+  }
+
+  authorizeAffiliationInvitation (isAuthorized, affiliationInvitationTodo): void {
+    AuthServices.authorizeAffiliationInvitation(
+      this.getAuthApiUrl,
+      this.getIdentifier,
+      affiliationInvitationTodo.affiliationId,
+      isAuthorized
+    )
+      .then()
+      .catch(err => {
+        // eslint-disable-line no-console
+        console.log('failed the call for authorization of affiliation invitation', err)
+      })
+  }
+
+  _buildTodoItemIfFromAffiliationInvitation (affiliationInvitation) {
+    const newTodo: TodoItemIF = {
+      draftTitle: undefined,
+      enabled: true,
+      filingId: -1, // not a filing
+      name: undefined,
+      order: 0,
+      subtitle: `From: ${affiliationInvitation.fromOrg.name}`,
+      status: FilingStatus.AFFILIATION_INVITATION_PENDING,
+      title: 'Request for authorization to mange this business',
+      affiliationInvitationDetails: {
+        id: affiliationInvitation.id,
+        fromOrgName: affiliationInvitation.fromOrg.name,
+        additionalMessage: affiliationInvitation.additionalMessage
+      }
+    }
+
+    return newTodo
+  }
+
+  /** Loads Org Affiliations invitation todo **/
+  async loadAffiliationInvitationsTodo () {
+    // feature-flag-it
+    if (!GetFeatureFlag('enable-affiliation-invitation-request-access')) {
+      return
+    }
+
+    // load all the invitations here and push them into todo items
+    const affiliationInvitations =
+      await AuthServices.fetchAffiliationInvitations(this.getAuthApiUrl, this.getIdentifier)
```

Comment:
This should be moved into the service itself, using await and .then afterwards seems fairly weird

Reply:
will update code a bit to be more clear